test_bst.py

```diff
@@ -75,6 +77,31 @@ def test_contains(filled_tree):
         assert t.contains(num) is False
 
 
+def test_in_order(filled_tree):
+    tree = filled_tree
+    for place, item in enumerate(tree.in_order()):
```

Comment:
If nothing at all comes out of `tree.in_order()`, then your assertion is never hit.  The test passes without proving anything at all.